### PR TITLE
feat: improve return type annotations for vim.api.*

### PR DESCRIPTION
**Pull request by lewis6991**

### CODE REVIEW

File: scripts/gen_eval_files.lua [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env -S nvim -l
+
```
Comment (dundargoc) [inline scripts/gen_eval_files.lua:2]: No: e0d97d264f83d45472e6cdd1051893db1083eefb

Comment (lewis6991) [reply scripts/gen_eval_files.lua:2]: The build scripts will still use the built binary. This is just so I can develop the script without invoking the cmake jet engine.

Comment (dundargoc) [reply scripts/gen_eval_files.lua:2]: Eh, fair, it should still work.

Comment (lewis6991) [reply scripts/gen_eval_files.lua:2]: I can confirm because `make doc` still works.

**Pull request merged**